src/extremeweatherbench/regions.py

```diff
@@ -0,0 +1,240 @@
+"""Region classes and utilities for the ExtremeWeatherBench package."""
+
+import dataclasses
+import logging
+from pathlib import Path
+from typing import Optional, Tuple, Union
+
+import geopandas as gpd  # type: ignore[import-untyped]
+import regionmask
+import xarray as xr
+from shapely import Polygon
+
+from extremeweatherbench import utils
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class Region:
+    """Base class for different region representations."""
+
+    def to_geopandas(self):
+        """Convert the region to a geopandas object."""
+        if isinstance(self, ShapefileRegion):
```

Comment:
Seeing how this is used later, why not just create a `geopandas` property? E.g.,

``` python
class Region:

    @property
    def geopandas(self):
        # construct GeoDataFrame here
        return gdf
```

Technically this would be an abstract property and each implementation would provide the logic.